client/assets/calendar.js

```diff
@@ -0,0 +1,121 @@
+import { saveAs } from 'file-saver'
+import Moment from 'moment'
+import {
+  extendMoment
+} from 'moment-range'
+
+export async function downloadCalendar({ plan }) {
+	const FileSaver = require('file-saver')
+	const moment = extendMoment(Moment)
+	const ics_day_list = ['SU', 'MO', 'TU', 'WE', 'TH', 'FR', 'SA'] //days of the week for ICS standard
+	const semester_year = (plan.courses[0].dates[0].slice(0, 2) < moment().month()) ? moment().year() + 1 : moment().year()
+
+	//required iCalendar info
+	//calendar_string is an array because what we use to print requires an array
+	let calendar_string = ['BEGIN:VCALENDAR\nPRODID:-//Nick and Pedram, Squaaaad//Course Calendar//EN\nVERSION:2.0\nCALSCALE:GREGORIAN']
+	calendar_string += '\nMETHOD:PUBLISH\nX-WR-CALNAME:' + plan.title
+	//timezone and daylight savings info
+	calendar_string += '\nX-WR-TIMEZONE:America/Los_Angeles\nBEGIN:VTIMEZONE\nTZID:America/Los_Angeles'
+	calendar_string += '\nX-LIC-LOCATION:America/Los_Angeles\nBEGIN:DAYLIGHT\nTZOFFSETFROM:-0800\nTZOFFSETTO:-0700\nTZNAME:PDT'
+	calendar_string += '\nDTSTART:19700308T020000\nRRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:-0700'
+	calendar_string += '\nTZOFFSETTO:-0800\nTZNAME:PST\nDTSTART:19701101T020000\nRRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU\nEND:STANDARD'
+	calendar_string += '\nEND:VTIMEZONE\n'
+
+	plan.courses.forEach(function(plan_course) {
+		//need to use moments to deal with the bug with the start day of the semester not being first day of the course
+		//start_moment = the starting date of the course and time the course starts each day
+		let start_moment = moment(plan_course.dates[0] + '/' + semester_year)
```

Comment:
Moment can parse from a specific format that you specify that would help to account for 24 hr time and not have to slice the string: [String + Format Constructor](https://momentjs.com/docs/#/parsing/string-format/)